code/obj/flock/structure/annihilator.dm

```diff
@@ -0,0 +1,133 @@
+
+		src.projectile_checkers = list()
+		for (var/turf/T as anything in turfs)
+			src.projectile_checkers += new /obj/annihilator_projectile_checker(T, src)
+
+		src.info_tag.set_info_tag("Not generating bolt")
+
+	building_specific_info()
+		return !src.powered ? "Not generating bolt." : GET_COOLDOWN(src, "bolt_gen_time") ? "Generation time left: [round(GET_COOLDOWN(src, "bolt_gen_time") / 10)] seconds." : "Bolt ready."
+
+	process(mult)
+		if (!src.flock)
+			if (src.powered)
+				src.update_flock_compute("remove")
+				if (src.checkers_powered)
+					src.power_projectile_checkers("off")
+			src.compute = 0
+			src.powered = FALSE
+		else if (src.flock.can_afford_compute(src.online_compute_cost))
+			src.compute = -src.online_compute_cost
+			if (!src.powered)
+				ON_COOLDOWN(src, "bolt_gen_time", 10 SECONDS)
+				src.update_flock_compute("apply")
+				src.powered = TRUE
+		else if (src.flock.used_compute > src.flock.total_compute() || !src.powered)
+			if (src.powered)
+				src.update_flock_compute("remove")
+				if (src.checkers_powered)
+					src.power_projectile_checkers("off")
+			src.compute = 0
+			src.powered = FALSE
+
+		if (src.powered)
+			if (GET_COOLDOWN(src, "bolt_gen_time"))
+				src.icon_state = "annihilator-generating"
+				if (src.checkers_powered)
+					src.power_projectile_checkers("off")
+			else
+				src.icon_state = "annihilator-ready"
+				if (!src.checkers_powered)
+					src.power_projectile_checkers("on")
+		else
+			src.icon_state = "annihilator-off"
+
+		src.info_tag.set_info_tag(!src.powered ? "Not generating bolt" : GET_COOLDOWN(src, "bolt_gen_time") ? "Generation time left: [round(GET_COOLDOWN(src, "bolt_gen_time") / 10)] seconds" : "Bolt ready")
+
+	proc/power_projectile_checkers(state)
+		var/on = state == "off" ? FALSE : TRUE
+		for (var/obj/annihilator_projectile_checker/checker as anything in src.projectile_checkers)
+			checker.on = on
+		src.checkers_powered = on
+
+	disposing()
+		for (var/obj/annihilator_projectile_checker/checker as anything in src.projectile_checkers)
+			qdel(checker)
+		..()
+
+
+/obj/annihilator_projectile_checker
+	name = null
+	desc = null
+	anchored = TRUE
+	density = FALSE
+	flags = UNCRUSHABLE
+	event_handler_flags = IMMUNE_SINGULARITY
+	invisibility = INVIS_ALWAYS
+	opacity = FALSE
+	mouse_opacity = 0
+
+	var/on = FALSE
+	var/obj/flock_structure/annihilator/connected_structure = null
+
+	New(turf/T, obj/flock_structure/annihilator/annihilator)
+		..()
+		src.connected_structure = annihilator
+
+	Cross(atom/movable/mover)
+		if (!src.on || !src.connected_structure)
+			return ..()
+		if (!istype(mover, /obj/projectile))
+			return ..()
+		var/obj/projectile/bullet = mover
+		if (!istype(bullet.proj_data, /datum/projectile/bullet))
+			return ..()
+		for (var/obj/flock_structure/annihilator/annihilator in view(src.connected_structure.checker_radius, src))
```

Comment:
why iterate through `view` if you already know which structure you're connected to?

Reply:
It's to make sure that the structure is actually in view, since the projectile checkers could appear behind walls

Reply:
ah makes sense, I was assuming it would go through walls